### PR TITLE
themis: publishable crates shouldn't depend on non-publishable ones

### PR DESCRIPTION
**Pull request by miraclx**
#7680 bumped the version of non-private crates to `0.15.0`. Unfortunately, the publish process failed, because `near-o11y` which is marked as a private package, is depended on by `near-vm-logic`, which is a non-private crate.

This adds a check to `themis`, to ensure we never get this kind of discrepancy merged into master.

Example Report:

CleanShot 2022-09-27 at 19 53 10@2x

### CODE REVIEW

File: core/o11y/LICENSE-APACHE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (matklad) [inline core/o11y/LICENSE-APACHE:1]: can we make these two a symbolic link? I think tahth should work

Comment (miraclx) [reply core/o11y/LICENSE-APACHE:1]: Ah yeah, could've sworn I did `ln`, apparently it was `cp`, lol. Fixing.

File: tools/themis/src/rules.rs [hunk 2]
```diff
@@ -289,3 +296,41 @@ pub fn publishable_has_near_link(workspace: &Workspace) -> anyhow::Result<()> {
...
+                        .entry(dep.parsed.manifest_path.clone())
+                        .or_insert_with(Vec::new)
+                        .push(pkg.parsed.name.clone())
+                }
+            }
+        }
+    }
+    if !outliers.is_empty() {
+        bail!(ComplianceError {
+            msg: "These private packages are depended on by publishable packages".to_string(),
```
Comment (frol) [inline tools/themis/src/rules.rs:317]: The original message was hard to understand at the first glance, so I tried to rewrite it. What do you think?

```suggestion
            msg: "These private crates break publishable crates. Either make these private crates publishable or avoid using them in the publishable crates.".to_string(),
```

**Review by frol — APPROVED** [review]
Overall, it is a great feature to prevent us going through this check manually! Thanks!

File: tools/themis/src/rules.rs (hunk 2 above)
Comment (miraclx) [reply tools/themis/src/rules.rs:317]: Yeah, this makes sense. I, too, went through a couple of description iterations, the above happened to be the shortest with most clarity, but I agree that we don't need to optimize for length, so more descriptive (and in this case, suggestive) is good. If you can submit a PR to this effect, that'd be awesome.

**Pull request merged**